flagsmith-engine/segments/evaluators.ts

```diff
@@ -55,18 +55,21 @@ function traitsMatchSegmentRule(
     );
 }
 
-function traitsMatchSegmentCondition(
+export function traitsMatchSegmentCondition(
     identityTraits: TraitModel[],
     condition: SegmentConditionModel,
     segmentId: number | string,
     identityId: number | string
 ): boolean {
-    if (condition.operator == PERCENTAGE_SPLIT) {
-        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(condition.value);
-    }
-
     const traits = identityTraits.filter(t => t.traitKey === condition.property_);
     const trait = traits.length > 0 ? traits[0] : undefined;
-
-    return trait ? condition.matchesTraitValue(trait.traitValue) : false;
-}
+    if (condition.operator == PERCENTAGE_SPLIT) {
+        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(<string>condition.value);
+    } else if (condition.operator === IS_SET ) {
+        return traits.length > 0
+    } else if (condition.operator === IS_NOT_SET){
+        return !(traits.length > 0)
```

Comment:
This is a bit convoluted. This evaluates the same as `traits.length == 0`, right?